### PR TITLE
Update dev

### LINKED ISSUE #3729: Merge 1.4.13 back to dev
1.4.13 is activated on mainnet, and there’s already a git tag on the main repo. We should merge back 1.4.13 changes to the dev branch before releasing 1.5. It should be possible to do it by cherry-picking individual PRs that went to 1.4.13 to avoid amount of merge conflicts, but we should be careful to not miss any changes.

### PR DESCRIPTION
**Pull request by Fraser999**
This PR brings the changes from `release-1.4.13` into `dev`.

There were many merge conflicts unfortunately, so it is worth reviewing all the changed files.  In particular, the costs in the `faucet_costs` test all reduced significantly, so it's worth checking changes related to that.

Closes #3729.

### CODE REVIEW

File: .drone.yml [hunk 1]
```diff
@@ -510,4 +510,4 @@ volumes:
   temp: {}
 
 trigger:
-  cron: [ nightly-tests-cron, feat-fast-sync-v2 ]
```
Comment (mpapierski) [inline .drone.yml:513]: This looks suspicious - we have this on dev branch https://github.com/casper-network/casper-node/blob/dev/.drone.yml#L513. We should preserve it?

File: execution_engine/Cargo.toml [hunk 2]
```diff
@@ -27,6 +27,7 @@ linked-hash-map = "0.5.3"
 lmdb-rkv = "0.14"
 log = { version = "0.4.8", features = ["std", "serde", "kv_unstable"] }
 num = { version = "0.4.0", default-features = false }
+num_cpus = "1"
```
Comment (mpapierski) [inline execution_engine/Cargo.toml]: It's not used in EE, we can remove this

File: execution_engine/src/core/engine_state/engine_config.rs [hunk 3]
```diff
@@ -20,6 +25,41 @@ const DAY_MILLIS: usize = 24 * 60 * 60 * 1000;
...
+/// 34 bytes for the public key + 10 bytes validator weight + 1 byte for delegation rate.
+const FIXED_SIZE_PER_VALIDATOR: u32 = 45;
+/// The size of a key of the `SeigniorageRecipientsSnapshot`, i.e. an `EraId`.
+const FIXED_SIZE_PER_ERA: u32 = U64_SERIALIZED_LENGTH as u32;
+/// The overhead of the Key::Hash under which the seigniorage snapshot lives.
+/// The hash length plus an additional byte for the tag.
+const KEY_HASH_SERIALIZED_LENGTH: u32 = KEY_HASH_LENGTH as u32 + 1;
+
+#[doc(hidden)]
+pub const fn compute_max_delegator_size_limit(
```
Comment (mpapierski) [inline execution_engine/src/core/engine_state/engine_config.rs]: Just to double check - we don't want to enforce max delegator size in 1.5 right? if so, we can remove this

File: execution_engine/src/storage/error/lmdb.rs [hunk 4]
```diff
@@ -11,10 +11,14 @@ use crate::storage::{error::in_memory, global_state::CommitError};
 #[derive(Debug, Clone, Error, PartialEq, Eq)]
 #[non_exhaustive]
 pub enum Error {
-    /// LMDB error returned from underlying `lmdb` crate.
+    /// Error from the underlying database.
```
Comment (mpapierski) [inline execution_engine/src/storage/error/lmdb.rs]:
```suggestion
    /// LMDB error returned from underlying `lmdb` crate.
```

File: execution_engine/src/storage/error/lmdb.rs [hunk 5]
```diff
@@ -11,10 +11,14 @@ use crate::storage::{error::in_memory, global_state::CommitError};
...
     #[error(transparent)]
     Lmdb(#[from] lmdb_external::Error),
 
+    /// Error when we cannot open a column family.
+    #[error("unable to open column family {0}")]
+    UnableToOpenColumnFamily(String),
+
```
Comment (mpapierski) [inline execution_engine/src/storage/error/lmdb.rs]: IIRC this is leftover from 1.4.x experimentation with rocksdb. We don't need this variant

```suggestion
```

File: execution_engine/src/storage/global_state/lmdb.rs [hunk 6]
```diff
@@ -26,6 +26,7 @@ use crate::{
 };
 
 /// Global state implemented against LMDB as a backing data store.
+#[derive(Clone)]
```
Comment (mpapierski) [inline execution_engine/src/storage/global_state/lmdb.rs]: It's small thing, but I don't remember we're actually using this Clone trait

File: node/src/components/event_stream_server.rs [hunk 7]
```diff
@@ -16,9 +16,6 @@
 //! This component uses a ring buffer for outbound events providing some robustness against
 //! unintended subscriber disconnects, if a disconnected subscriber re-subscribes before the buffer
 //! has advanced past their last received event.
-//!
-//! For details about the SSE model and a list of supported SSEs, see:
-//! <https://github.com/CasperLabs/ceps/blob/master/text/0009-client-api.md#rpcs>
```
Comment (mpapierski) [inline node/src/components/event_stream_server.rs:21]: I'm not sure about this one, so I'm leaving this comment just to highlight it for you. This link is valid, and the comment is present on dev branch

File: .drone.yml (hunk 1 above)
Comment (Fraser999) [reply .drone.yml:513]: I discussed this diff with @TomVasile and he was happy that we just retain the one branch here going forward.

File: node/src/components/event_stream_server.rs (hunk 7 above)
Comment (Fraser999) [reply node/src/components/event_stream_server.rs:21]: Yeah, we did decide to remove links to CEPS from the rustdocs, so this one isn't an error :)

File: execution_engine/Cargo.toml (hunk 2 above)
Comment (Fraser999) [reply execution_engine/Cargo.toml]: Fixed in [a901201](https://github.com/casper-network/casper-node/pull/3754/commits/a90120169c847f0938a237dce6362d1b2879c325).

File: execution_engine/src/core/engine_state/engine_config.rs (hunk 3 above)
Comment (Fraser999) [reply execution_engine/src/core/engine_state/engine_config.rs]: Fixed in [a901201](https://github.com/casper-network/casper-node/pull/3754/commits/a90120169c847f0938a237dce6362d1b2879c325).

File: execution_engine/src/storage/error/lmdb.rs (hunk 4 above)
Comment (Fraser999) [reply execution_engine/src/storage/error/lmdb.rs]: Fixed in [a901201](https://github.com/casper-network/casper-node/pull/3754/commits/a90120169c847f0938a237dce6362d1b2879c325).

File: execution_engine/src/storage/error/lmdb.rs (hunk 5 above)
Comment (Fraser999) [reply execution_engine/src/storage/error/lmdb.rs]: Fixed in [a901201](https://github.com/casper-network/casper-node/pull/3754/commits/a90120169c847f0938a237dce6362d1b2879c325).

File: execution_engine/src/storage/global_state/lmdb.rs (hunk 6 above)
Comment (Fraser999) [reply execution_engine/src/storage/global_state/lmdb.rs]: Fixed in [a901201](https://github.com/casper-network/casper-node/pull/3754/commits/a90120169c847f0938a237dce6362d1b2879c325).

**Review by mpapierski — APPROVED** [review]

**Comment by Fraser999** [comment]
bors merge

**Pull request merged**